test/functional/tests/io/test_write_fetch.py

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+import uuid
+from api.cas import casadm
+from api.cas.cache_config import CacheMode, CacheLineSize, CacheModeTrait
+from core.test_run import TestRun
+from storage_devices.disk import DiskTypeSet, DiskTypeLowerThan, DiskType
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils.os_utils import Udev
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrize("cache_mode", [CacheMode.WO, CacheMode.WB, CacheMode.WT])
+@pytest.mark.parametrize("cache_line_size", CacheLineSize)
+def test_write_fetch_full_misses(cache_mode, cache_line_size):
+    """
+        title: No caching of full write miss operations with block size smaller than cache line size
+        description: |
+          Validate CAS ability to not cache entire cache line size for full write miss operations
+          when block size is smaller than cache line size – no fetch for writes
+        pass_criteria:
+          - Appropriate number of write full misses and writes to cache in cache statistics
+          - Appropriate number of writes to cache in iostat
+    """
+    file_size = Size(300, Unit.MebiByte)
+    validated_stat_name = "write full misses"
+
+    with TestRun.step("Start cache and add core."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        cache = casadm.start_cache(cache_disk, cache_mode, cache_line_size)
+        Udev.disable()
+        core = cache.add_core(core_disk)
+    with TestRun.step("Run writes to CAS device using fio."):
+        io_stats_before_io = cache_disk.get_io_stats()
+        run_fio(core.system_path,
+                f"write:{int((cache_line_size.value / 2).get_value(Unit.KibiByte))}k",
```

Comment:
Please make operation type separate parameter from its size. For setting operation type use values from enum. Operation size could be type of Size() instead of string